lib/money/currency.rb

```diff
@@ -32,6 +34,23 @@ class UnknownCurrency < ArgumentError; end
 
     class << self
 
+      def new(id)
+        id = id.to_s.downcase
+        unless stringified_keys.include?(id)
+          raise UnknownCurrency, "Unknown currency '#{id}'"
+        end
+
+        instance(id.to_s.downcase) || super
```

Comment:
Also seems like there should be `super(id)` to use string, when it was called with sym.